### PR TITLE
feat: Update period validation for requests

### PR DESCRIPTION
**Pull request by MWO1024**
# Description

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/dotnet/wholesale-api/Edi.UnitTests/Validators/AggregatedTimeSeriesRequest/AggregatedTimeSeriesRequestValidatorTests.cs [hunk 1]
```diff
@@ -22,21 +23,31 @@
 using Energinet.DataHub.Wholesale.Edi.Validation.Helpers;
 using FluentAssertions;
 using Microsoft.Extensions.DependencyInjection;
+using Moq;
 using NodaTime;
 using Xunit;
 
 namespace Energinet.DataHub.Wholesale.Edi.UnitTests.Validators.AggregatedTimeSeriesRequest;
 
+[SuppressMessage(
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi.UnitTests/Validators/AggregatedTimeSeriesRequest/AggregatedTimeSeriesRequestValidatorTests.cs:32]: Delete?

File: source/dotnet/wholesale-api/Edi.UnitTests/Validators/AggregatedTimeSeriesRequest/AggregatedTimeSeriesRequestValidatorTests.cs [hunk 2]
```diff
@@ -77,8 +88,126 @@ public async Task Validate_WhenPeriodSizeIsInvalid_ReturnsUnsuccessfulValidation
...
+
+        // Act
+        var validationErrors = await _sut.ValidateAsync(request);
+
+        // Assert
+        validationErrors.Should().BeEmpty();
+    }
+
+    [Fact]
+    public async Task
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi.UnitTests/Validators/AggregatedTimeSeriesRequest/AggregatedTimeSeriesRequestValidatorTests.cs:138]: :(

File: source/dotnet/wholesale-api/Edi/Validation/AggregatedTimeSeriesRequest/Rules/PeriodValidationRule.cs [hunk 3]
```diff
@@ -22,12 +22,14 @@ public class PeriodValidationRule(PeriodValidationHelper periodValidationHelper)
...
+    private readonly PeriodValidationHelper _periodValidationHelper = periodValidationHelper;
 
     private static readonly ValidationError _invalidDateFormat = new("Forkert dato format for {PropertyName}, skal være YYYY-MM-DDT22:00:00Z eller YYYY-MM-DDT23:00:00Z / Wrong date format for {PropertyName}, must be YYYY-MM-DDT22:00:00Z or YYYY-MM-DDT23:00:00Z", "D66");
     private static readonly ValidationError _invalidWinterMidnightFormat = new("Forkert dato format for {PropertyName}, skal være YYYY-MM-DDT23:00:00Z / Wrong date format for {PropertyName}, must be YYYY-MM-DDT23:00:00Z", "D66");
     private static readonly ValidationError _invalidSummerMidnightFormat = new("Forkert dato format for {PropertyName}, skal være YYYY-MM-DDT22:00:00Z / Wrong date format for {PropertyName}, must be YYYY-MM-DDT22:00:00Z", "D66");
-    private static readonly ValidationError _startDateMustBeLessThen3Years = new("Dato må max være 3 år tilbage i tid / Can maximum be 3 years back in time", "E17");
+    private static readonly ValidationError _startDateMustBeLessThen3Years = new("Dato må max være 3 år og 6 måneder tilbage i tid / Can maximum be 3 years and 6 months back in time", "E17");
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi/Validation/AggregatedTimeSeriesRequest/Rules/PeriodValidationRule.cs:32]:
```suggestion
    private static readonly ValidationError _startDateMustBeLessThen3Years = new("Dato må max være {_allowedTimeFrameYearsFromNow} år og {_allowedTimeFrameMonthsFromNow} måneder tilbage i tid / Can maximum be {_allowedTimeFrameYearsFromNow} years and {_allowedTimeFrameMonthsFromNow} months back in time", "E17");
```

File: source/dotnet/wholesale-api/Edi/Validation/Helpers/PeriodValidationHelper.cs [hunk 4]
```diff
@@ -18,42 +18,47 @@ namespace Energinet.DataHub.Wholesale.Edi.Validation.Helpers;
...
     }
 
-    public bool IsMonthOlder3Years2Months(Instant periodStart)
+    public bool IsMonthOfDateOlderThanXYearsAndYMonths(Instant periodStart, int years, int months)
     {
-        var zonedDateTime = new ZonedDateTime(periodStart, dateTimeZone);
-        var zonedCurrentDataTime = new ZonedDateTime(clock.GetCurrentInstant(), dateTimeZone);
-        var threeYearsAndTwoMonthsAgo = zonedCurrentDataTime.LocalDateTime.PlusYears(-3).PlusMonths(-2);
+        var dateInQuestion = periodStart.InZone(_dateTimeZone);
+        var threeYearsAndTwoMonthsAgo = _clock.GetCurrentInstant()
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi/Validation/Helpers/PeriodValidationHelper.cs:52]:
```suggestion
        var xYearsAndYMonthsAgo = _clock.GetCurrentInstant()
```

Comment (RasmusGraabaek) [reply source/dotnet/wholesale-api/Edi/Validation/Helpers/PeriodValidationHelper.cs:52]: Or "thresholdDate". I dont know

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/PeriodValidationRule.cs [hunk 5]
```diff
@@ -94,7 +99,7 @@ public Task<IList<ValidationError>> ValidateAsync(DataHub.Edi.Requests.Wholesale
 
     private void MustNotBe3YearsAnd3MonthsOld(Instant periodStart, ICollection<ValidationError> errors)
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/PeriodValidationRule.cs:100]: New name

File: source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/PeriodValidationRule.cs [hunk 6]
```diff
@@ -18,17 +18,22 @@
...
+
     private static readonly ValidationError _invalidDateFormat =
         new(
             "Forkert dato format for {PropertyName}, skal være YYYY-MM-DDT22:00:00Z eller YYYY-MM-DDT23:00:00Z / Wrong date format for {PropertyName}, must be YYYY-MM-DDT22:00:00Z or YYYY-MM-DDT23:00:00Z",
             "D66");
 
     private static readonly ValidationError _startDateMustBeLessThanOrEqualTo3YearsAnd3Months =
         new(
-            "Der kan ikke anmodes om data for 3 år og 3 måneder tilbage i tid / It is not possible to request data 3 years and 3 months back in time",
+            "Der kan ikke anmodes om data for 3 år og 6 måneder tilbage i tid / It is not possible to request data 3 years and 6 months back in time",
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi/Validation/WholesaleServicesRequest/Rules/PeriodValidationRule.cs:36]: Can we, as in the timeseries case, have a local variable containing the values for our thresholds?

**Comment by MWO1024** [comment]
We made the changes in EDI instead.

**Pull request closed without merging**